File: programminglanguages/Assignment3/task2.scala
//Author: Thomas Curtis
//Description: The program ask for a file from the user. It checks the contens of 
//	the file for key words and then outputs statements about the file

import scala.io.StdIn.readLine
import java.io.File
import java.io.PrintWriter
import scala.io.Source
import scala.util.matching.Regex

object task2 {

	def main(args: Array[String]):Unit = {
		
		println("Enter file name")
		var file = scala.io.StdIn.readLine() //input from user

		for(line <- Source.fromFile(file).getLines) {   //gets each line from file
			
			val cplus = line.contains("cplus")
			val ruby  = line.contains("ruby")
			val haskell = line.contains("haskell")
			val scala = line.contains("scala")

			if(cplus || ruby) {
				println("This file context is okay")
			} else if (haskell || scala) {
				println("The file is great")	
			} else {
				println("The file is uninteresting")
			}
		}
    }
}

File: programminglanguages/task2.rb
#Author: Thomas Curtis
#Program 2: The program searches the context of the given file and then tells if it is a valuable file

#Input for file name
puts "Enter file name"
fileName = gets.chomp;

#opening and reading the file
File.open(fileName) do |file|

    while line = file.gets

       if line.chomp == "Python" || line.chomp == "Ruby"
            puts "This file is important"
        
        elsif line.chomp == "Scala"
            puts "The file is interesting"
        
        elsif line.chomp == "Haskell"
            puts "The file is valuable"
        
        else 
            puts "The file is worthless"
        end
        
    end

end
